File: backend/src/main/java/br/ufpr/sept/so2/modules/presenca/infrastructure/persistence/PresencaJpaRepository.java
package br.ufpr.sept.so2.modules.presenca.infrastructure.persistence;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.UUID;

public interface PresencaJpaRepository extends JpaRepository<PresencaJpaEntity, UUID> {

    boolean existsByEventoIdAndUsuarioIdAndFase(UUID eventoId, UUID usuarioId, String fase);

    boolean existsByEventoIdAndDeviceUuidAndUsuarioIdNot(UUID eventoId, String deviceUuid, UUID usuarioId);

    List<PresencaJpaEntity> findByEventoIdAndUsuarioId(UUID eventoId, UUID usuarioId);

    long countByEventoId(UUID eventoId);
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/presenca/infrastructure/persistence/PresencaJpaRepository.kt
package br.ufpr.sept.so2.modules.presenca.infrastructure.persistence

import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface PresencaJpaRepository : JpaRepository<PresencaJpaEntity, UUID> {
    fun existsByEventoIdAndUsuarioIdAndFase(eventoId: UUID, usuarioId: UUID, fase: String): Boolean

    fun existsByEventoIdAndDeviceUuidAndUsuarioIdNot(eventoId: UUID, deviceUuid: String, usuarioId: UUID): Boolean

    fun findByEventoIdAndUsuarioId(eventoId: UUID, usuarioId: UUID): List<PresencaJpaEntity>

    fun countByEventoId(eventoId: UUID): Long
}
